### DISCUSSION TITLE
path to taxdumps

### DISCUSSION THREAD

**Discussion by gregdenay** [bio-raum/FooDMe2#6] [General]
hi @marchoeppner 
The workflow currently crashed with an error in `filter taxonomy` module:
the path to nodes.dmp that is given to the helper script is only `nodes.dmp`, missing the path before.

I tried to look th eplace where this is defined, is `conf/ressources.config` the right place? I looks alright there.
Check the last commit

**Comment by marchoeppner** [comment]
@gregdenay 
Runs through all the way to SEQTABLE_TO_FASTA for me - with both singularity and conda, on your last commit (which is missing a JSON output - this is coming, I guess?). 

Otherwise, yes, the path is specified in conf/ressources.config

For me, the process directory looks as follows (ls work/\*/\*/taxonomy.json)

``
insgesamt 7572
drwxr-xr-x. 2 marc marc    4096 24. Jun 10:32 ./
drwxr-xr-x. 4 marc marc      82 24. Jun 10:33 ../
-rw-r--r--. 1 marc marc       0 24. Jun 10:32 .command.begin
-rw-r--r--. 1 marc marc       0 24. Jun 10:32 .command.err
-rw-r--r--. 1 marc marc       0 24. Jun 10:32 .command.log
-rw-r--r--. 1 marc marc       0 24. Jun 10:32 .command.out
-rw-r--r--. 1 marc marc   11126 24. Jun 10:32 .command.run
-rw-r--r--. 1 marc marc     316 24. Jun 10:32 .command.sh
-rw-r--r--. 1 marc marc     270 24. Jun 10:32 .command.trace
-rw-r--r--. 1 marc marc       1 24. Jun 10:32 .exitcode
lrwxrwxrwx. 1 marc marc      56 24. Jun 10:32 merged.dmp -> /home/marc/references/foodme2/1.0/new_taxdump/merged.dmp
lrwxrwxrwx. 1 marc marc      55 24. Jun 10:32 nodes.dmp -> /home/marc/references/foodme2/1.0/new_taxdump/nodes.dmp
lrwxrwxrwx. 1 marc marc      63 24. Jun 10:32 rankedlineage.dmp -> /home/marc/references/foodme2/1.0/new_taxdump/rankedlineage.dmp
-rw-r--r--. 1 marc marc 7718145 24. Jun 10:32 taxonomy.json
-rw-r--r--. 1 marc marc      69 24. Jun 10:32 versions.yml

`` 
Can you check what that looks like for you?

**Comment by gregdenay** [comment]
```
ls work/*/*/*.dmp
work/fd/5e1c99de043eabc31fe305aac8d91c/merged.dmp  work/fd/5e1c99de043eabc31fe305aac8d91c/nodes.dmp  work/fd/5e1c99de043eabc31fe305aac8d91c/rankedlineage.dmp
```
Also sollte es eigentlich da sein, richtig?

`nextflow run main.nf -profile test -c ~/.nextflow/nextflow.config --reference_base /home/debian/NGS/nextflow_refs/`
Ergibt:
```
ERROR ~ Error executing process > 'FOODME2:BLAST_TAXONOMY:HELPER_FILTER_TAXONOMY (Taxdump)'

Caused by:
  Process `FOODME2:BLAST_TAXONOMY:HELPER_FILTER_TAXONOMY (Taxdump)` terminated with an error exit status (1)


Command executed:

  filter_taxonomy.py --nodes nodes.dmp \
  --rankedlineage rankedlineage.dmp \
  --merged merged.dmp \
  --taxid 32524 \
  --json taxonomy.json

  cat <<-END_VERSIONS > versions.yml
  "FOODME2:BLAST_TAXONOMY:HELPER_FILTER_TAXONOMY":
      python3: $(python3 --version  | sed -e "s/Python //")
  END_VERSIONS

Command exit status:
  1

Command output:
  (empty)

Command error:
  Traceback (most recent call last):
    File "/home/debian/NGS/spezies_indev/FooDMe2/bin/filter_taxonomy.py", line 25, in <module>
      main(args.nodes,
    File "/home/debian/NGS/spezies_indev/FooDMe2/bin/filter_taxonomy.py", line 19, in main
      tax = taxidTools.read_taxdump(nodes, lineage, merged)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/home/debian/NGS/nextflow_envs/format_midori-3f0e915d93dbb32cf63994cb2731c2f5/lib/python3.12/site-packages/taxidTools/factories.py", line 40, in read_taxdump
      for line in _parse_dump(nodes):
    File "/home/debian/NGS/nextflow_envs/format_midori-3f0e915d93dbb32cf63994cb2731c2f5/lib/python3.12/site-packages/taxidTools/factories.py", line 109, in _parse_dump
      with open(filepath, 'r') as dmp:
           ^^^^^^^^^^^^^^^^^^^
  FileNotFoundError: [Errno 2] No such file or directory: 'nodes.dmp'

Work dir:
  /home/debian/NGS/spezies_indev/FooDMe2/work/ work/fd/5e1c99de043eabc31fe305aac8d91c
```

**Comment by marchoeppner** [comment]
mach mal:

ls -lh work/\*/\*/\*.dmp

damit er den ganzen Pfad der symlinks anzeigt - und dann prüfe, ob die Datei am Ende des Symlinks wirklich existiert. Eventuell ist deine Installation unvollständig. 

Alternativ:

ls -lh /home/debian/NGS/nextflow_refs/foodme2/1.0/new_taxdump/*.dmp

**Reply by gregdenay** [reply]
ja tatsächlich war die taxdump nicht da ...
Jetzt läufts

**Comment by gregdenay** [comment]
.
